experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -470,23 +464,146 @@ export interface Sequence<TTypes extends AllowedTypes> extends TreeField {
 	 */
-	boxedAt(index: number): TypedNodeUnion<TTypes>;
+	indexOf(searchElement: UnboxNodeUnion<TTypes>, fromIndex?: number): number;
 
 	/**
-	 * Calls the provided callback function on each child of this sequence, and returns an array that contains the results.
-	 * @param callbackfn - A function that accepts the child, its index, and this field.
+	 * Determines whether an sequence includes a certain element, returning true or false as appropriate.
+	 * @param searchElement - The element to search for.
+	 * @param fromIndex - The index at which to begin the search. If fromIndex is omitted, the search starts at index 0.
+	 */
+	includes(searchElement: UnboxNodeUnion<TTypes>, fromIndex?: number): boolean;
+
+	/**
+	 * Iterates the indices of the elements in the sequence.
+	 */
+	keys(): IterableIterator<number>;
+
+	/**
+	 * Iterates the values in the sequence.
+	 */
+	values(): IterableIterator<UnboxNodeUnion<TTypes>>;
+
+	/**
+	 * Iterates the elements in the sequence as tuples of index and element.
 	 */
-	map<U>(callbackfn: (value: UnboxNodeUnion<TTypes>, index: number, array: this) => U): U[];
+	entries(): IterableIterator<[number, UnboxNodeUnion<TTypes>]>;
+
+	/**
+	 * Executes a provided function once per each element in the sequence.
+	 * @param callbackFn - The function to run for each element.
+	 */
+	forEach(
+		callbackFn: (element: UnboxNodeUnion<TTypes>, index: number, sequence: this) => void,
+	): void;
+
+	/**
+	 * Returns the value of the first element in the sequence where `predicate` is true, and undefined otherwise.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	find(
+		predicate: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => boolean,
+	): UnboxNodeUnion<TTypes> | undefined;
+	find<T extends UnboxNodeUnion<TTypes>>(
+		predicate: (element: UnboxNodeUnion<TTypes>, index: number, sequence: this) => element is T,
+	): T | undefined;
+
+	/**
+	 * Returns the value of the last element in the sequence where `predicate` is true, and undefined otherwise.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	findLast(
+		predicate: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => boolean,
+	): UnboxNodeUnion<TTypes> | undefined;
+	findLast<T extends UnboxNodeUnion<TTypes>>(
+		predicate: (element: UnboxNodeUnion<TTypes>, index: number, sequence: this) => element is T,
+	): T | undefined;
+
+	/**
+	 * Returns the index of the first element in the sequence where `predicate` is true, otherwise -1.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	findIndex(
+		predicate: (value: UnboxNodeUnion<TTypes>, index: number, sequence: this) => boolean,
+	): number;
+
+	/**
+	 * Returns the index of the last element in the sequence where `predicate` is true, otherwise -1.
+	 * @param predicate - The predicate to run on each element.
+	 */
+	findLastIndex(
```

Comment:
Rather than find index, I think findBoxed would be better. It solves a lot of issues (undefined being returned not being clear if its a value or not since it won't be a valid value, no need to use -1 as a setinal, and you can just get the index by looking at the parent of the node you found if needed.

Reply:
Also the versions which are boxed are strictly more capable. I think we should either only have the boxed version, or have both. Only having the unboxed version means that sometimes you have to port your code away from using these helpers if you need access to thr parent, or the type field or will want to do editing on the thing you found.

Reply:
Also for consistancy, everything in this APi surface is either boxed, or has both. Thus you can tell if an API does unboxing by seeing that there is a boxed version. This PR breaks that.

Reply:
Isn't it more consistent to be align with the core reading API behavior? When getting an item from the tree, we will return boxed nodes for non-values, and unboxed primitives for values. I think this API should do the same. We don't expose the concept of boxed values anywhere else. So far.

Reply:
Regarding boxing, the handful of problems solved by boxing have not been significant enough to outweigh the customer's negative emotional response when being presented with multiple API modes.  We should continue to look hard for alternative solutions (or as in this case, accept the status quo) before reintroducing boxing in the spec.